src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -360,6 +401,25 @@ public void postDealFireDamage(CallbackInfo callbackInfo) {
         }
     }
 
+    @Inject(method = "onStruckByLightning", at = @At(value = "FIELD", target = LIGHTNING_DAMAGESOURCE_FIELD, opcode = Opcodes.GETSTATIC))
+    public void preLightning(EntityLightningBolt lightningBolt, CallbackInfo callbackInfo) {
+        if (!this.world.isRemote) {
+            this.originalLightning = DamageSource.LIGHTNING_BOLT;
+            DamageSource.IN_FIRE = new EntityDamageSource("lightningBolt", lightningBolt);
```

Comment:
Mainly did this to pass the entity (the lightning bolt) through as a source, which has a representation in Sponge.